### ISSUE TITLE
How to initialize build2 in an existing project?

### ISSUE THREAD

**Issue by alexchandel** [build2/build2#111]
I have a collection of subprojects (a couples libraries & executables that use them) that must be linked against two 3rd-party precompiled libraries/headers, and I need to cross-compile these projects from macos to linux & windows.

* How is one supposed to initialize build2 in an existing project?

`bdep new -l c++ -t bare` fails with these errors:
```
warning: unable to extract project summary from /Users/alex/RA/README.md
  info: using generic summary in manifest
error: unable to create /Users/alex/RA/.gitignore: file exists
```

Just jumping in and trying `bdep new --source -t lib -l c++ Library_One` results in
```
warning: /Users/alex/RA/ does not look like a package directory
created new library source subdirectory Library_One in /Users/alex/RA/Library_One/
```

Deleting & recreating my own `.gitignore` (which shouldn't be), later attempting to add with `bdep new --source -t exe -l c++ Client_One` and build results in the errors like below.
```
error: no rule to update Client_One/exe{Client_One}
  info: re-run with --verbose=4 for more information
  info: while applying rule alias to update dir{Client_One/}
  info: while applying rule alias to update dir{./}
info: failed to update dir{./}
```

**Comment by boris-kolpackov** [comment]
You can use hooks to manage your existing files like `.gitignore`. You probably want to merge yours and `bdep`-generated and there is an example of doing that in the [`bdep-new`](https://build2.org/bdep/doc/bdep-new.xhtml) man page. Specifically, the following sequence of steps works for me:

```
git init
echo '*.foo' >.gitignore

bdep new -l c++ -t bare --pre-hook  "mv .gitignore .gitignore.bak" --post-hook "cat .gitignore.bak >>.gitignore" --post-hook "rm .gitignore.bak"
bdep new --source -t lib -l c++ Library_One
bdep new --source -t exe -l c++ Client_One

bdep init -C @gcc cc config.cxx=g++
b
```

**Comment by alexchandel** [comment]
That's what I ended up doing, but automatic modification of the `.gitignore` would be nice.  It's already checked into version control, so there's no danger of messing anything up.

**Comment by boris-kolpackov** [comment]
Yes, automatically merging existing `.gitignore` is probably a good idea. I don't think we should do it silently (there is no guarantee that the current version is actually committed). Instead, we can fail with a suggestion of an option to request merging.

We also generate `.gitattributes` but that one I think we should just skip generating (with a warning) if one already exists.
